src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java

```diff
@@ -2659,4 +2660,30 @@ public void setActiveCell(CellAddress address) {
         _sheet.setActiveCellRow(row);
         _sheet.setActiveCellCol(col);
     }
+    private void shiftFormulas(int startRow, int endRow, int n, boolean forRowElseColumnShift){ 
+        int sheetIndex = _workbook.getSheetIndex(this);
+        String sheetName = _workbook.getSheetName(sheetIndex);
+        short externSheetIndex = _book.checkExternSheet(sheetIndex);
+        FormulaShifter shifter = FormulaShifter.createForRowShift(
+                         externSheetIndex, sheetName, startRow, endRow, n, SpreadsheetVersion.EXCEL97);
+        _sheet.updateFormulasAfterCellShift(shifter, externSheetIndex);
+
+        int nSheets = _workbook.getNumberOfSheets();
+        for (int i = 0; i < nSheets; i++) {
+            InternalSheet otherSheet = _workbook.getSheetAt(i).getSheet();
+            if (otherSheet == this._sheet) {
+                continue;
+            }
+            short otherExtSheetIx = _book.checkExternSheet(i);
+            otherSheet.updateFormulasAfterCellShift(shifter, otherExtSheetIx);
+        }
+        _workbook.getWorkbook().updateNamesAfterCellShift(shifter); 
+    } 
+
+    public void shiftColumns(int startColumn, int endColumn, int n){ 
```

Comment:
Add a Javadoc to indicate what features are shifted and what isn't (yet)
Add a `@Beta` annotation to indicate this is still being developed.
Add `@Override`